### PR TITLE
Add support for EIP-1559

### PR DESCRIPTION
**Pull request by mfornet**

### CODE REVIEW

File: eth2near/ethashproof/ethash/consensus.go [hunk 1]
```diff
@@ -570,14 +572,14 @@ func (ethash *Ethash) Finalize(chain consensus.ChainReader, header *types.Header
 	header.Root = state.IntermediateRoot(chain.Config().IsEIP158(header.Number))
 
 	// Header seems complete, assemble into a block and return
-	return types.NewBlock(header, txs, uncles, receipts), nil
+	return types.NewBlock(header, txs, uncles, receipts, trie.NewStackTrie(nil)), nil
```
Comment (sept-en) [inline eth2near/ethashproof/ethash/consensus.go:575]: why do we add `trie.NewStackTrie(nil)` here?

**Review by sept-en — APPROVED** [review]
LGTM!

File: eth2near/ethashproof/ethash/consensus.go (hunk 1 above)
Comment (mfornet) [reply eth2near/ethashproof/ethash/consensus.go:575]: We are using a newer version of go-ethereum and this is a new required argument.

**Review by sept-en — APPROVED** [review]
LGTM!

NB: this should be merged only after #602 and #615 are merged.

**Pull request merged**